include/flexible.hxx

```diff
@@ -0,0 +1,448 @@
+
+/*!************************************************************************
+ * This file is autogenerated - see  flexible.hxx.in.py
+ **************************************************************************/
+
+#pragma once
+
+#ifndef __FLEXIBLE_H__
+#define __FLEXIBLE_H__
+
+#include <bout_types.hxx>
+#include <bout/deprecated.hxx>
+//class Field2D;
+//class Field3D;
+#include <field2d.hxx>
+#include <field3d.hxx>
+#include <field_data.hxx>
+#include <bout/dataiterator.hxx>
+#include <boutexception.hxx>
+
+const char * strLocation(CELL_LOC);
+const Field2D interp_to(const Field2D&,CELL_LOC);
+const Field3D interp_to(const Field3D&,CELL_LOC);
+
+/// Template for having one field at different locations. If a Field
+/// is not yet known for that location, it will be created and
+/// cached. It is further possible to provide the staggered fields, if
+/// it is e.g. crated from an analytical expression.
+template <typename F>
+class Flexible: public FieldData{
+  typedef unsigned int uint;
+public:
+  Flexible(const F & main){
+    init(new F(main));
+  };
+  template <typename... Args>
+  Flexible(Args... args) {
+    F * main = new F(args...);
+    init(main);
+  }
+  F & getNonConst(CELL_LOC loc){
+    return *((F*)geti(loc));
+  }
+  /// Get a const reference of the field at the specific location. If
+  /// the CELL_LOC is CELL_DEFAULT the mainlocation will be returned.
+  const F & get(CELL_LOC loc){
+    return * geti(loc);
+  };
+  Flexible<F> & operator=(const F& f) {
```

Comment:
I think this might be one reason your clang build fails (possibly) -- you're not actually returning anything from this routine (and other nearby ones) despite the signature saying that you will. I've seen this cause issues with clang elsewhere.

Reply:
Thanks, that helped indeed :+1: